CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/parameters/types/ElementTypeParameterValue.java

```diff
@@ -82,4 +86,26 @@ protected ParameterValue createCopy() {
     public String toString() {
         return elementType == null ? "No Value" : elementType.getShortLabel();
     }
+
+    @Override
+    public int hashCode() {
+        return 41 * 5 + Objects.hashCode(this.elementType);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final ElementTypeParameterValue other = (ElementTypeParameterValue) obj;
+        return this.elementType == other.elementType;
+    }
```

Comment:
Hey @Nova-2119 what change in your PR required the introduction of the `hash` and `equals` methods? Is it the call to `instanceof` ?

Reply:
SingloeChoiceParameterValue.setChoiceData(ParameterValue) uses the .contains method so for any ParameterValue to work properly with it we need to override equals.
Previously this was causing issues because two identical ElementTypeParamaterValues weren't considered equal as they weren't the same object.